File: spare_parts_shop_flutter/ping_server.dart
import 'dart:async';
import 'dart:io';

void main() async {
  final url = 'https://inventorymanagement-afhl.onrender.com/api'; // Your base URL
  const pingInterval = Duration(minutes: 5);
  
  print('🟢 Ping service started - Pinging every 5 mins');
  
  // Initial ping
  await pingServer(url);
  
  // Set up periodic pings
  Timer.periodic(pingInterval, (timer) async {
    await pingServer(url);
  });
  
  // Keep the program alive
  await Completer<void>().future;
}

Future<void> pingServer(String url) async {
  try {
    final client = HttpClient();
    final request = await client.getUrl(Uri.parse(url));
    final response = await request.close();
    
    // Only log errors or very quiet success
    await response.drain();
    client.close();
  } catch (e) {
    print('🔴 Ping failed: ${e.toString().substring(0, 50)}...');
  }
}

File: spare-parts-shop-frontend/ping-server.js
const http = require('http');
const https = require('https');

const url = 'https://inventorymanagement-afhl.onrender.com/api'; // Your Render backend URL
const intervalMs = 5 * 60 * 1000; // 5 minutes in milliseconds

console.log('🚀 Starting ping service...');
console.log(`📡 Pinging ${url} every 5 minutes to keep Render awake`);
console.log('--------------------------------------------------------');

// Function to ping server
function pingServer() {
  const protocol = url.startsWith('https') ? https : http;
  
  const req = protocol.get(url, (res) => {
    const timestamp = new Date().toISOString().replace('T', ' ').substring(0, 19);
    console.log(`✅ [${timestamp}] Ping successful - Status: ${res.statusCode}`);
    res.resume(); // Consume response body to complete the request
  });
  
  req.on('error', (e) => {
    const timestamp = new Date().toISOString().replace('T', ' ').substring(0, 19);
    console.error(`❌ [${timestamp}] Ping failed: ${e.message}`);
  });
  
  req.setTimeout(10000, () => {
    req.destroy();
    const timestamp = new Date().toISOString().replace('T', ' ').substring(0, 19);
    console.error(`❌ [${timestamp}] Ping failed: Request timed out`);
  });
}

// Initial ping
pingServer();

// Set up interval
setInterval(pingServer, intervalMs);

// Keep process alive
process.stdin.resume();
